tools/analysis/runnable.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited, Google, and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from collections import namedtuple
+import trappy
+import argparse
+
+EventData = namedtuple('EventData', ['time', 'event', 'data'])
+LatData = namedtuple('LatData', ['pid', 'switch_data', 'wake_data', 'last_wake_data', 'running', 'wake_pend', 'latency', 'lat_total'])
+
+parser = argparse.ArgumentParser(description='Analyze runnable times')
+
+parser.add_argument('--trace', dest='trace_file', action='store', required=True,
+                    help='trace file')
+
+parser.add_argument('--rt', dest='rt', action='store_true', default=False,
+                    help='only consider RT tasks')
+
+parser.add_argument('--normalize', dest='normalize', action='store_true', default=False,
+                    help='normalize time')
+
+parser.add_argument('--rows', dest='nrows', action='store', default=20, type=int,
+                    help='normalize time')
+
+parser.add_argument('--total', dest='lat_total', action='store_true', default=False,
+                    help='sort by total runnable time')
+
+parser.add_argument('--start-time', dest='start_time', action='store', default=0, type=float,
+                    help='trace window start time')
+
+parser.add_argument('--end-time', dest='end_time', action='store', default=None, type=float,
+                    help='trace window end time')
+
+args = parser.parse_args()
+
+path_to_html = args.trace_file
+rtonly = args.rt
+nrows = args.nrows
+
+# Hash table of pid -> LatData named tuple
+latpids = {}
+
+def switch_cb(data):
```

Comment:
*_cb functions should take latpids as a parameter to avoid wild accesses to the latpids global.
functools.partial can be used to tie latpids and the function when setting up the callbacks:
https://docs.python.org/2/library/functools.html#functools.partial